src/util/matchToKartoffel.js

```diff
@@ -343,6 +345,155 @@ const match_nv_sql = (obj, dataSource) => {
+                (rawKey === "lastName") ? null : delete obj[rawKey];
+                break;
+            //rank
+            case fn[dataSource].rank:
+                obj.rank = obj[rawKey];
+                (rawKey === "rank") ? null : delete obj[rawKey];
+                break;
+            // dischargeDay
+            case fn[dataSource].dischargeDay:
+                obj.dischargeDay = obj[rawKey];
+                (rawKey === "dischargeDay") ? null : delete obj[rawKey];
+                break;
+            // clearance
+            case fn[dataSource].clearance:
+                obj.clearance = obj[rawKey];
+                (rawKey === "clearance") ? null : delete obj[rawKey];
+                break;
+            // currentUnit
+            case fn[dataSource].currentUnit:
+                obj.currentUnit = obj[rawKey];
+                (rawKey === "currentUnit") ? null : delete obj[rawKey];
+                break;
+            // serviceType
+            case fn[dataSource].serviceType:
+                obj.serviceType = obj[rawKey];
+                (rawKey === "serviceType") ? null : delete obj[rawKey];
+                break;
+            //mobilePhone
+            case fn[dataSource].mobilePhone:
+                validators().mobilePhone.test(obj[rawKey]) ? obj.mobilePhone = [obj[rawKey]] : delete obj[rawKey];
+                (rawKey === "mobilePhone") ? null : delete obj[rawKey];
+                break;
+            //address
+            case fn[dataSource].address:
+                obj.address = obj[rawKey];
+                (rawKey === "address") ? null : delete obj[rawKey];
+                break;
+            //mail
+            case fn[dataSource].mail:
+                obj.mail = obj[rawKey];
+                (rawKey === "mail") ? null : delete obj[rawKey];
+                break;
+            //job
+            case fn[dataSource].job:
+                obj.job = obj[rawKey];
+                (rawKey === "job") ? null : delete obj[rawKey];
+                break;
+            case fn[dataSource].profession:
+                if (!obj[fn[dataSource].job]) {
+                    obj.job = obj[rawKey];
+                }
+
+                delete obj[rawKey];
+                break;
+            //hierarchy
+            case fn[dataSource].hierarchy:
+                let hr = obj[rawKey].replace('\\', '/');
+                if (hr.includes('/')) {
+                    hr = hr.split('/').map(unit => unit.trim());
+
+                    for(const [index, value] of hr.entries()) {
+                        if (isStrContains(value, [`${obj[fn[dataSource].firstName]} ${obj[fn[dataSource].lastName]}`, '-', ''])) {
+                            hr.splice(index);
+                            break;
+                        }
+                    }
+
+                    hr = hr.join('/');
+                }
+
+                obj.hierarchy = `${fn.rootHierarchy.city}${company ? '/' + company : ''}${hr ? '/' + hr : ''}`;
```

Comment:
use with the variable above 
:arrow_double_up:

Reply:
👍